backend/globaleaks/utils/users_details_filter.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8
+import re
+
+class UserDetailsFilter:
+    """
+    Filters the text for any user details.
+    Removes things like uuid4, email, pgp formatted strings, session ids.
+    """
+
+    def __init__(self, text):
+        """
+        Filters or scrubs the given text.
+        Arguments:
+            text {string} -- text to be fileted.
+        """
+        self._text = text
+
+    def remove_bgp_formatted_string(self):
+        # remove public key block
+        self._text = re.sub(r'-----BEGIN PGP PUBLIC KEY BLOCK-----.*-----END PGP PUBLIC KEY BLOCK-----',
+            "*****", self._text, 0, re.DOTALL)
+
+        # remove private key block
+        self._text = re.sub(r'-----BEGIN PGP PRIVATE KEY BLOCK-----.*-----END PGP PRIVATE KEY BLOCK-----',
+            "*****", self._text, 0, re.DOTALL)
+
+
+    def remove_session_id_details(self):
+        # remove session id details
+        # here the assumption is that json output session details
+        # may look something like this:
+        #   {...., "session_id": .... } or  {...., "session_id": .... , .... }
+
+        self._text = re.sub(r'"session_id":.*,', '"***":***,', self._text, 0, re.DOTALL)
+        self._text = re.sub(r'"session_id":.*}', '"***":***}', self._text, 0, re.DOTALL)
```

Comment:
probably this is duplicated?

better still to be explicit with a "session_id:filtered" so we know what was filtered

Reply:
Thanks for the comment. I've updated the regex string and removed the code which was filtering twice.